python/lsst/meas/algorithms/loadReferenceObjects.py

```diff
@@ -296,26 +297,23 @@ def _makeBoxRegion(BBox, wcs, BBoxPadding):
         outerLocalBBox.grow(BBoxPadding)
         innerLocalBBox.grow(-1*BBoxPadding)
 
-        # Handle the fact that the inner bounding box shrunk to a zero sized region in at least one
-        # dimension, in which case all reference catalogs must be checked fully against the input
-        # bounding box
+        # Handle the case where the inner bounding box shrunk to a zero sized
+        # region in at least one dimension, in which case all reference catalogs
+        # must be checked fully against the input bounding box.
         if innerLocalBBox.getDimensions() == geom.Extent2D(0, 0):
-            innerSkyRegion = sphgeom.Box()
-        else:
-            innerBoxCorners = innerLocalBBox.getCorners()
-            innerSphCorners = [wcs.pixelToSky(corner).getVector() for corner in innerBoxCorners]
-
-            innerSkyRegion = sphgeom.ConvexPolygon(innerSphCorners)
+            innerLocalBBox = geom.Box2D(BBox)
```

Comment:
Two things here confuse me:
 - This comment says "in at least one dimension" both before and after these changes, but the code clearly says "in both dimensions".
 - Did you remove the else clause because we're _actually_ hitting the case where the bbox is zero-sized in both dimensions _and_ converting an empty `Box2D` to a `sphgeom.Box` does not produce an empty `sphgeom.Box`?  Both of those clauses being true (where the first seems unlikely, and the latter seems like a separate bug) seem to be the only way the removal of the `else` clause could affect what happens.

Or have I just misread the logic here?

Reply:
Yes, the wording and implementation of this function really confused me too.  In playing around with the `BBoxPadding` that gets passed in, I went to the extreme (as changing it wasn’t seeming to have an effect on the trimming of the catalog), I tried a value of 2500.  This should, in principle, only shrink one dimension of an HSC chip down to zero...but I did hit this clause (maybe we don’t allow an single-dimension zero without setting the other to zero as well...that seems to be sensible, actually) and then hit an error when the code tried to return `innerSphCorners` as it did not get defined in that clause.  Regardless of dimensions, it seemed setting the inner box to a NaN-box was not a good thing to do, so I set it back to the original bbox (I _think_ this is in line with the wording!) and that way the other settings could be moved out of the clause, thus avoiding the error.

As a side note, I’m not clear on the usefulness of this shrunken bbox, but I may be missing “the point”.

Reply:
> we don’t allow an single-dimension zero without setting the other to zero as well

Exactly right, and I should have thought of this earlier.  But the comment is certainly confusing if you don't know that.  Mind fixing it here?

> Regardless of dimensions, it seemed setting the inner box to a NaN-box was not a good thing to do, so I set it back to the original bbox (I think this is in line with the wording!) and that way the other settings could be moved out of the clause, thus avoiding the error.

Ah, ok, what I had missed is that you are setting this to the original bbox, and not to the empty one.  Still a weird way to get out of this edge case, but let's not mess with it now.  Mind commenting to that effect, too?  Something like
```
# This probably isn't the right way to handle an empty inner bbox, but it's what calling code currently expects.
```

Reply:
Ok, I've reworded, but this last comment has me thinking more about my side note above.  I don't really understand the value and use of this "inner" -- sure to be contained within the detectors actual sky -- bounding box. 
 It is used to skip the reference cat trimming loop if the reference catalog lies entirely within this inner region, which does makes sense...but if this case is true, it means the reference catalog loaded does not cover anywhere near what is needed/being asked for (by a huge factor in area since it is shrinking by the amount we really want it to be grown by!  I've actually also added a `log.warn` if that triggers since the user should at least be aware of this situation...) , but if the main goal is to have a box that is almost certain to be contained within the true WCS region of the dataset, we could set it to "shrink by max of BBoxPadding or N% of the number of pixels in the shorter dimension".

Reply:
I don't think we want that warning in `_filterFunction` - I think that'll cause it to warn if any one of the reference catalog shards is within the region requested, rather than warning only if the region covered by _all_ of the shards is within the region requested.  And that's why the logic makes sense.

That said, I don't think we actually have any reference catalogs whose shard size is small enough for this to happen, but it's hypothetically possible.  That raises the question of whether this is unnecessary complexity, because it's optimizing a case that doesn't happen in practice, but I think that's probably out of scope for this ticket.

Reply:
Ah, ok..now I think I understand the logic (I wasn’t thinking about this per-shard).  I totally agree on all of the above and will get rid of the warn!